packages/dev/serializers/src/glTF/2.0/glTFMaterialExporter.ts

```diff
@@ -84,240 +65,89 @@ function getFileExtensionFromMimeType(mimeType: ImageMimeType): string {
 }
 
 /**
- * Utility methods for working with glTF material conversion properties.  This class should only be used internally
- * @internal
+ * Computes the metallic factor.
+ * Exported to be testable in unit tests.
+ * @param diffuse diffused value
+ * @param specular specular value
+ * @param oneMinusSpecularStrength one minus the specular strength
+ * @returns metallic value
  */
-export class _GLTFMaterialExporter {
-    /**
-     * Represents the dielectric specular values for R, G and B
-     */
-    private static readonly _DielectricSpecular: Color3 = new Color3(0.04, 0.04, 0.04);
-
-    /**
-     * Allows the maximum specular power to be defined for material calculations
-     */
-    private static readonly _MaxSpecularPower = 1024;
-
-    /**
-     * Mapping to store textures
-     */
-    private _textureMap: { [textureId: string]: ITextureInfo } = {};
-
-    // Mapping of internal textures to images to avoid exporting duplicate images.
-    private _internalTextureToImage: { [uniqueId: number]: { [mimeType: string]: Promise<number> } } = {};
-
-    /**
-     * Numeric tolerance value
-     */
-    private static readonly _Epsilon = 1e-6;
-
-    /**
-     * Reference to the glTF Exporter
-     */
-    private _exporter: _Exporter;
-
-    constructor(exporter: _Exporter) {
-        this._textureMap = {};
-        this._exporter = exporter;
-    }
-
-    /**
-     * Specifies if two colors are approximately equal in value
-     * @param color1 first color to compare to
-     * @param color2 second color to compare to
-     * @param epsilon threshold value
-     * @returns boolean specifying if the colors are approximately equal in value
-     */
-    private static _FuzzyEquals(color1: Color3, color2: Color3, epsilon: number): boolean {
-        return Scalar.WithinEpsilon(color1.r, color2.r, epsilon) && Scalar.WithinEpsilon(color1.g, color2.g, epsilon) && Scalar.WithinEpsilon(color1.b, color2.b, epsilon);
+export function solveMetallic(diffuse: number, specular: number, oneMinusSpecularStrength: number): number {

```

Comment:
Question: I exported this function because it's used in an integration test (from back when it was a static method of GLTFMaterialExporter), but it feels wrong to have done. Anyone have a preference?

Reply:
Marked internal and renamed to _SolveMetallic. Kept export.